### DISCUSSION TITLE
Why can not transpile all the node modules by default ?

### DISCUSSION THREAD

**Discussion by ra1nj** [callstack/repack#732] [Q&A] [answered]
Since Metro transpile all the node modules by default, we don't have to add any additional config . Can we do this in webpack ?  i tried to add a rule like this :  /node_modules\/(.*)/,  but it will cause a error : Property 'exports' doesn't exist

Can someone tell me why is this happening ?

**Comment by jbroma** [comment, marked as answer]
Hey @ra1nj,

that's a very good question! The template we provide with Re.Pack, limits the number of transformed modules by default to not be too hard on the performance side. 

The error you've encountered is due to how modules are processed by both bundlers - some rules from `@react-native/babel-preset` don't play nicely in webpack. You usually need to track down these modules and adjust them on a case-by-case basis. In the case you have above, it's most likely that the module was recognized as an ESM, but was then compiled down to CJS (because of the RN preset) and webpack didn't not take that into account and forgot to expose `exports` property for it. That's the gist of it 

In the next major version of Re.Pack (V5 with support for Rspack), the performance impact is mostly gone thanks to the `swc` under the hood. We've also worked to create a default set of rules that will allow to get the right result out of the box instead of spending hours configuring the rules like now. I'm well aware how frustrating issues like this are and we're trying to solve it entirely.

**Reply by ra1nj** [reply]
Wow, thank you for this detailed explanation, looking forward for the new version!
